app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt

```diff
@@ -60,13 +68,109 @@ class LoginSSOViewModel @Inject constructor(
     navigationManager,
     clientScopeProviderFactory,
     authServerConfigProvider,
-    userDataStoreProvider
+    userDataStoreProvider,
+    coreLogic
 ) {
 
     var openWebUrl = MutableSharedFlow<String>()
 
     fun login() {
         loginState = loginState.copy(ssoLoginLoading = true, loginError = LoginError.None).updateSSOLoginEnabled()
+
+        loginState.userInput.text.also {
+            if (validateEmailUseCase(it)) {
+                domainLookupFlow()
+            } else {
+                ssoLoginWithCodeFlow()
+            }
+        }
+    }
+
+    fun onCustomServerDialogDismiss() {
+        loginState = loginState.copy(customServerDialogState = null)
+    }
+
+    fun onCustomServerDialogConfirm() {
+        viewModelScope.launch {
+            if (loginState.customServerDialogState != null) {
+                authServerConfigProvider.updateAuthServer(loginState.customServerDialogState!!.serverLinks)
+
+                val authScope = coreLogic.versionedAuthenticationScope(loginState.customServerDialogState!!.serverLinks)().let {
+                    when (it) {
+                        is AutoVersionAuthScopeUseCase.Result.Failure.Generic,
+                        AutoVersionAuthScopeUseCase.Result.Failure.TooNewVersion,
+                        AutoVersionAuthScopeUseCase.Result.Failure.UnknownServerVersion -> {
+                            loginState = loginState.copy(customServerDialogState = null)
+                            return@launch
+                        }
+
+                        is AutoVersionAuthScopeUseCase.Result.Success -> {
+                            it.authenticationScope
+                        }
+                    }
+                }
+
+                authScope.ssoLoginScope.fetchSSOSettings().also {
+                    when (it) {
+                        is FetchSSOSettingsUseCase.Result.Failure -> {}
+                        is FetchSSOSettingsUseCase.Result.Success -> {
+                            it.defaultSSOCode?.let { ssoCode ->
```

Comment:
Can we cover error cases with test as well ? when `AutoVersionAuthScopeUseCase` `fetchSSOSettings` fail